lib/eventFactory.js

```diff
@@ -453,50 +455,88 @@ async function convertStages(config) {
             }
         });
 
+        newStage.eventId = eventId;
         newStage.jobIds = jobIds;
+
         delete newStage.jobs; // extra field from yaml parser
 
+        // Check for setup and teardown
+        const setupJobName = `${STAGE_PREFIX}${key}:setup`;
+        const teardownJobName = `${STAGE_PREFIX}${key}:teardown`;
+        const setupJobExists = Object.keys(jobs).find(j => j === setupJobName || j === `~${setupJobName}`);
+        const teardownJobExists = Object.keys(jobs).find(j => j === teardownJobName || j === `~${teardownJobName}`);
+
+        if (setupJobExists) {
```

Comment:
How do we handle the use case, when `setup` job exists but the steps has been updated in the config?

Reply:
Same for `teardown`

Reply:
Please ignore if we are not allowing `setup`/`teardown` to be declared in the configuration in the initial version.

Reply:
I guess `setup` and `teardown` jobs would already be created/updated during PR sync logic.
With that assumption, do we need this check?
Will there be a scenario where `setup` job does not exist for a stage?

Reply:
hmm should at least be an empty object